File: pyg_parent/pyg_content/pyg_content_interface/src/main/java/com/pinyougou/content/service/ContentService.java
package com.pinyougou.content.service;

import com.pinyougou.pojo.TbContent;
import entity.PageResult;
import entity.ResultBean;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ContentService {

	/**
	 * 增加
	*/
	public void add(TbContent content);
	
	
	/**
	 * 修改
	 */
	public void update(TbContent content);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public ResultBean findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbContent content, int pageNum, int pageSize);

	ResultBean findAllContentCategory();


    ResultBean findAll();

    ResultBean findAllContent(Long categoryId);
}

File: pyg_parent/pyg_manager_web/src/main/webapp/admin/js/service/contentService.js
//服务层
app.service('contentService',function($http){


	this.findAllContentCategory = function () {
		return $http.get('../content/findAllContentCategory')
    };

	this.findAll = function () {
        return $http.get('../content/findAll')
    };

	//查询实体
	this.findOne=function(id){
		return $http.get('../content/findOne/'+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../content/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../content/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../content/delete/'+ids);
	}
	//搜索
	this.search=function(pageNum,pageSize,searchEntity){
		return $http.post('../content/search/'+pageNum+"/"+pageSize, searchEntity);
	}    	
});
